packages/widgetbook/lib/src/addons/grid_addon/grid_painter.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+
+class GridPainter extends CustomPainter {
+  GridPainter({
+    required this.horizontalDistance,
+    required this.verticalDistance,
+  });
+
+  final int horizontalDistance;
+  final int verticalDistance;
```

Comment:
We can use a single field called `dimension`, and use an `assert` here to prevent negative/zero values. It's better than using an `if` check that will not notify the user that something is wrong.

```suggestion
  GridPainter(
    this.dimension,
  ) : assert(dimension > 0);

  final int dimension;
```

Reply:
I think we should let the dimension begin from 0 as a dimension of 0 means no grid.

Reply:
I am personally not a fan of silent failing, because it has a bad DX especially for packages.
Creating a grid with "zero" dimension should not be possible, as it results in no grid at all.
My philosophy is, if the user ordered bananas and oranges, we should give them bananas and oranges, but we cannot give them only bananas and say nothing about the oranges.

Reply:
Alright.